File: Testio/app/src/main/java/com/azzdorfrobotics/android/testio/di/AppComponent.java
package com.azzdorfrobotics.android.testio.di;

import com.azzdorfrobotics.android.testio.App;
import com.azzdorfrobotics.android.testio.network.NetworkModule;
import com.azzdorfrobotics.android.testio.preferences.AppPreferences;
import com.azzdorfrobotics.android.testio.preferences.PreferenceModule;
import dagger.Component;
import javax.inject.Singleton;

/**
 * Created on 25.12.2017.
 *
 * @author Mykola Tychyna (iMykolaPro)
 */
@Singleton
@Component(modules = {
    AppModule.class,
    ResourceModule.class,
    NetworkModule.class,
    PreferenceModule.class,
    RealmModule.class})
public interface AppComponent {

    void inject(App app);

    ServerComponent serverComponent();

    AppPreferences provideAppPreferences();
}

File: app/src/main/java/lt/havefun/tesonetfunparty/injections/components/AppComponent.kt
package lt.havefun.tesonetfunparty.injections.components

import dagger.Component
import lt.havefun.tesonetfunparty.App
import lt.havefun.tesonetfunparty.injections.modules.ApiModule
import lt.havefun.tesonetfunparty.injections.modules.AppModule
import lt.havefun.tesonetfunparty.injections.modules.PreferencesModule
import lt.havefun.tesonetfunparty.injections.modules.RoomModule
import javax.inject.Singleton

@Singleton
@Component(modules = [
    AppModule::class,
    ApiModule::class,
    PreferencesModule::class,
    RoomModule::class
])
interface AppComponent {
    fun inject(application: App)
}
